lib/getenv.c

```diff
@@ -27,25 +27,46 @@
 
 #include "memdebug.h"
 
-static
-char *GetEnv(const char *variable)
+static char *GetEnv(const char *variable)
 {
 #if defined(_WIN32_WCE) || defined(CURL_WINDOWS_APP)
   (void)variable;
   return NULL;
-#else
-#ifdef WIN32
-  char env[4096];
-  char *temp = getenv(variable);
-  env[0] = '\0';
-  if(temp != NULL)
-    ExpandEnvironmentStringsA(temp, env, sizeof(env));
```

Comment:
Not sure whether dropping the expansion should be done for BC reasons.

Reply:
Having that dual purpose in the function did not make sense to me. Since I wanted to deduplicate I would have to split it off into ExpandEnv. I passed on writing that after reading the doc for ExpandEnvironmentStringsA because it doesn't tell you if all the environment variables you passed it have been expanded. Even if you compare size you can't be sure if there are multiple. We could loop it like `begin = strchr(begin, '%') if(!begin) break; ++begin; end = strchr(begin, '%'); if(!end) error;` but I just don't think it's worth the extra code right now for the one instance. We use it for `%USERPROFILE%\\Application Data` so I changed that call to basically `env = curl_getenv(USERPROFILE); aprintf("%s\\Application Data", env);`